### PR TITLE
Create cluster-api.sigs.k8s.io to publish documentation with Netlify

### PR DESCRIPTION
**Pull request by davidewatson**
Please see https://github.com/kubernetes/community/issues/3454#issuecomment-484534141 for more context.

### CODE REVIEW

File: dns/zone-configs/k8s.io.yaml [hunk 1]
```diff
@@ -240,3 +240,7 @@ velodrome:
 kind.sigs:
   type: CNAME
   value: k8s-kind.netlify.com.
+# https://github.com/kubernetes-sigs/cluster-api docs (@dwat, @jdetiber, @@justinsb)
+cluster-api.sigs:
+  type: CNAME
+  value: k8s-cluster-api.netlify.com.
```
Comment (mrbobbytables) [inline dns/zone-configs/k8s.io.yaml]: Would `kubernetes-sigs-cluster-api.netlify.com` be an okay alternative? That should align with the project name to be created in netlify.

Comment (davidewatson) [reply dns/zone-configs/k8s.io.yaml]: Done.

File: dns/zone-configs/k8s.io.yaml [hunk 2]
```diff
@@ -240,3 +240,7 @@ velodrome:
 kind.sigs:
   type: CNAME
   value: k8s-kind.netlify.com.
+# https://github.com/kubernetes-sigs/cluster-api docs (@dwat, @jdetiber, @@justinsb)
+cluster-api.sigs:
+  type: CNAME
+  value: kubernetes-sigs-cluster-api.netlify.com
```
Comment (cblecker) [inline dns/zone-configs/k8s.io.yaml]: Trailing period needed. Also, please squash.

Comment (davidewatson) [reply dns/zone-configs/k8s.io.yaml]: Fixed. Thanks!

**Comment by davidewatson** [comment]
Ping @cblecker. Is there anything else that needs to be done before this issue can be closed out? I would like to begin publishing the Cluster API GitBook next week, now that many of us have returned from KubeCon...

**Comment by cblecker** [comment]
Is the netlify site live? I don't see it.

**Comment by mrbobbytables** [comment]
It looks like the [cluster-api repo](https://github.com/kubernetes-sigs/cluster-api) doesn't have a `netlify.toml` config  in the root of the directory. 😬

**Comment by davidewatson** [comment]
It was my understanding that the FQDN needed to exist before I could configure Netlify. Are we saying I should first get my cluster-api build script committed first?

My apologies if I have the process out of order...

**Comment by mrbobbytables** [comment]
No worries, you have it right following the directions. If you add a Netlify config you can view the site on it's netlify address (`kubernetes-sigs-cluster-api.netlify.com`) to preview the site until the DNS name is configured. 

The reason for the request going in first would be to work through any possible name issues. In reality they can be done in tandem. I can update the instructions to reflect that a bit better.

**Comment by davidewatson** [comment]
Thanks @mrbobbytables. I'll go ahead and finish the `cluster-api` testing. Thanks again for your help!

**Comment by davidewatson** [comment]
In order to test the generated site before the cluster-api.sigs.k8s.io domain exists, I temporarily disabled the redirects in the `netlify.toml`.

Here is the site: https://kubernetes-sigs-cluster-api.netlify.com/

**Comment by davidewatson** [comment]
In order to test the [`netlify.toml`](https://github.com/kubernetes-sigs/cluster-api/pull/659/commits/fa4b2167abbf94dac88bca92a92e013fec04fc44) I had to comment out the redirects to the (currently) non-existent cluster-api.sigs.k8s.io domain.

If we merge this PR first, then I can add the redirects back before merging https://github.com/kubernetes-sigs/cluster-api/pull/659. Alternatively, I can merge the PR, then follow up on this PR, and finally, submit a second PR to fix up the redirects... Can you recommend how I should proceed?

And thanks so much and so many times to everyone [1] who has helped make this happen. Our SIG is excited to have continuously deployed documentation. It will greatly increase the velocity with which we can improve the documentation! :)

[1] @mrbobbytables, @BenTheElder, @chenopis, @cblecker, et al.

**Comment by cblecker** [comment]
I'll merge and apply this.

/lgtm
/approve

**Comment by cblecker** [comment]
@davidewatson This is now pushed and live.

**Pull request merged**